src/main/scala/io/github/evis/scalafix/maven/plugin/params/SourceDirectoryParam.scala

```diff
@@ -1,18 +1,21 @@
 package io.github.evis.scalafix.maven.plugin.params
 
-import java.nio.file.Paths
+import java.io.File
 
 object SourceDirectoryParam {
 
-  def apply(sourceDirectory: String): MojoParam = {
-    _.withPaths(List(getCanonicalPath(sourceDirectory)).filter(_.toFile.exists))
+  def apply(dirs: Iterable[File]): MojoParam = {
+    _.withPaths(dirs.flatMap(getCanonicalPath).toList.distinct)
   }
 
+  def apply(dirs: String*): MojoParam = apply(dirs.map(getFile))
+
   // Usually we execute plugin on source directory with path either like
   // src/{main,test}/java/../scala or src/{main,test}/scala/../scala. Anyway,
   // this path is relative, not canonical. Scalafix throws MissingTextDocument
   // exception when tries to find sources by relative path. As a workaround, we
   // provide canonical path for Scalafix.
-  private def getCanonicalPath(directory: String) =
-    Paths.get(directory).toFile.getCanonicalFile.toPath
+  private def getCanonicalPath(directory: File) =
+    if (directory.exists()) Some(directory.getCanonicalFile.toPath) else None
```

Comment:
IMO, if the source directory chosen by the plugin user doesn't exist, we should throw an exception instead of hiding it from the user. Otherwise, users will spend a lot of time trying to find, why their sources aren't checked by scalafix.